tests/extension/extension_test.php

```diff
@@ -0,0 +1,318 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace local_sitsgradepush;
+
+defined('MOODLE_INTERNAL') || die();
+global $CFG;
+require_once($CFG->dirroot . '/local/sitsgradepush/tests/fixtures/tests_data_provider.php');
+
+/**
+ * Tests for the extension class.
+ *
+ * @package    local_sitsgradepush
+ * @copyright  2024 onwards University College London {@link https://www.ucl.ac.uk/}
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @author     Alex Yeung <k.yeung@ucl.ac.uk>
+ */
+final class extension_test extends \advanced_testcase {
+
+    /** @var \stdClass $course1 Default test course 1 */
+    private \stdClass $course1;
+
+    /** @var \stdClass Default test student 1 */
+    private \stdClass $student1;
+
+    /** @var \stdClass Default test assignment 1 */
+    private \stdClass $assign1;
+
+    /** @var \stdClass Default test quiz 1*/
+    private \stdClass $quiz1;
+
+    /**
+     * Set up the test.
+     *
+     * @return void
+     */
+    protected function setUp(): void {
+        parent::setUp();
+        $this->resetAfterTest();
+
+        // Set admin user.
+        $this->setAdminUser();
+
+        // Get data generator.
+        $dg = $this->getDataGenerator();
+
+        // Setup testing environment.
+        set_config('late_summer_assessment_end_' . date('Y'), date('Y-m-d', strtotime('+2 month')), 'block_lifecycle');
```

Comment:
https://moodledev.io/docs/4.4/apis/core/clock might be useful

Reply:
Thanks, but seems not applicable here.